docs/gcp/README.md

```diff
@@ -0,0 +1,89 @@
+# Running a node on GCP
+
+## GCP Setup
+
+We use the [Google Cloud CLI](https://cloud.google.com/sdk/docs/install) to deploy the nodes.
+
+If you have already a gcloud project, you can simply use it. Otherwise, or if prefer, you will need to create one:
+
+```bash
+gcloud projects create <unique project id>
+gcloud config set project <unique project id>
+```
+
+We use Compute Engine to run the node. First, we set the region and zone where we want to deploy (see [regions/zones](https://cloud.google.com/compute/docs/regions-zones)).
+For example:
+
+```bash
+gcloud config set compute/region europe-west1
+gcloud config set compute/zone europe-west1-b
+```
+
+We also create a firewall rule for opening the node port (9454 by default)
+
+```bash
+gcloud compute firewall-rules create starknet-node \
+  --allow tcp:9545 \
+  --target-tags starknet-mainnet,starknet-goerli
+```
+
+## Deployment
+
+You first need to load then env variables defined in the .env file. For example: `export $(xargs <.env)`
+
+Then run the following command to pop a mainnet node
+(note that only the `PATHFINDER_ETHEREUM_API_URL_MAINNET` makes this node a mainnet one, the other "mainnet" are just naming):
+
+```bash
+gcloud compute instances create-with-container starknet-mainnet \
+  --container-image=eqlabs/pathfinder:latest \
+  --container-restart-policy on-failure \
+  --container-env PATHFINDER_ETHEREUM_API_URL=${PATHFINDER_ETHEREUM_API_URL_MAINNET} \
+  --create-disk name=mainnet-data \
+  --container-mount-disk mount-path="/usr/share/pathfinder/data",name=mainnet-data \
+  --tags starknet-mainnet
+```
+
+Note that in the previous command, we explicitely create a `disk`. If you kill your node at some point and want to pop another container,
+you can reuse the same `disk` by replace `--create-dist` with `--disk`. You can check your disks with `gcloud compute disks list`.
+
+Then, log into the host machine using ssh to update the disk permission (beware of the naming of the disk, see previous comment):
+
+```bash
+gcloud compute ssh starknet-mainnet
+$ sudo lsblk
+$ sudo chmod a+w /mnt/disks/gce-containers-mounts/gce-persistent-disks/mainnet-data
```

Comment:
Why is this step required?

Reply:
This comes from the [doc "Add persistent disk"](https://cloud.google.com/compute/docs/disks/add-persistent-disk). The first steps of this doc (creating the disk) are performed in the `create-with-container` command using the flag `create-disk`.

However, I have to say that I'm not a GCP expert so there may possibly something better to do